src/workers/parser/mapBindings.js

```diff
@@ -90,12 +122,13 @@ export default function mapExpressionBindings(
       return;
     }
 
-    if (hasDestructuring(node)) {
+    if (hasNestedDestructuring(node)) {
       shouldUpdate = false;
       return;
     }
 
-    if (!t.isForStatement(parent.node)) {
+    // the isMapped check is to avoid processing the newly added nodes
+    if (!t.isForStatement(parent.node) && !isMapped) {
```

Comment:
Happy to rename/add new variable or slightly adjust the way we iterate.
Currently, the newly inserted/replaced nodes are iterated again, so
`let __decl1__ = x;` is transformed to `self.__decl1__ = x`;

Reply:
Moved the condition to the top. I suppose we will want to adjust it, as the check does might be confusing.